extension/app/src/lib/conversation.ts

```diff
@@ -265,3 +265,50 @@ export async function postMessage({
 
   return new Ok(await mRes.json());
 }
+
+export async function retryMessage({
+  owner,
+  conversationId,
+  messageId,
+}: {
+  owner: LightWorkspaceType;
+  conversationId: string;
+  messageId: string;
+}): Promise<Result<{ message: UserMessageWithRankType }, SubmitMessageError>> {
+  const token = await getAccessToken();
+  const user = await getStoredUser();
+
+  if (!user) {
+    // This should never happen.
+    return new Err({
+      type: "user_not_found",
+      title: "User not found.",
+      message: "Please log in again.",
+    });
+  }
+
+  const mRes = await fetch(
```

Comment:
you can use `dustAPI.request()` instead  :-p

Reply:
will do after you merged your PR with the tooling 🙈